### PR TITLE
Update `on-color` tokens to use `on-fill` tokens

### PR DESCRIPTION
**Pull request by sophschneider**
Fixes https://github.com/Shopify/polaris/issues/10765

Didn't do `Button` file as that should be solved in https://github.com/Shopify/polaris/issues/10353

### CODE REVIEW

File: polaris-react/src/components/Autocomplete/components/MappedOption/MappedOption.scss [hunk 1]
```diff
@@ -13,7 +13,6 @@
 
   svg {
     fill: var(--p-color-icon);
-    color: var(--p-color-icon-on-color);
```
Comment (sophschneider) [inline polaris-react/src/components/Autocomplete/components/MappedOption/MappedOption.scss:16]: These aren't doing anything, the icons are already on a white background (the autocomplete popover surface)

File: polaris-react/src/components/CheckableButton/CheckableButton.scss [hunk 2]
```diff
@@ -25,7 +25,7 @@
   height: 100%;
 
   svg {
-    fill: var(--p-color-icon-on-color);
+    fill: var(--p-color-text-brand-on-bg-fill);
```
Comment (sophschneider) [inline polaris-react/src/components/CheckableButton/CheckableButton.scss:28]: background is black brand color

File: polaris-react/src/components/Frame/components/Toast/Toast.scss [hunk 3]
```diff
@@ -28,7 +28,7 @@ $Backdrop-opacity: 0.88;
 
 .Action {
   margin-left: var(--p-space-400);
-  color: var(--p-color-text-on-color);
+  color: var(--p-color-text-inverse);
```
Comment (sophschneider) [inline polaris-react/src/components/Frame/components/Toast/Toast.scss:31]: other text was also updated to `inverse` in this pr https://github.com/Shopify/polaris/issues/10386

File: polaris-react/src/components/RadioButton/RadioButton.scss [hunk 4]
```diff
@@ -30,7 +30,7 @@
       border-color: var(--p-color-border-interactive);
 
       &::before {
-        background-color: var(--p-color-icon-on-color);
+        background-color: var(--p-color-text-brand-on-bg-fill);
```
Comment (sophschneider) [inline polaris-react/src/components/RadioButton/RadioButton.scss:33]: background is brand color

File: polaris-react/src/components/Autocomplete/components/MappedOption/MappedOption.tsx [hunk 5]
```diff
@@ -19,6 +19,7 @@ export const MappedOption = memo(function MappedOption({
   selected,
   singleSelection,
 }: MappedOption) {
+  console.log('hi');
```
Comment (laurkim) [inline polaris-react/src/components/Autocomplete/components/MappedOption/MappedOption.tsx]: 🔥

Comment (sophschneider) [reply polaris-react/src/components/Autocomplete/components/MappedOption/MappedOption.tsx]: oopsies

**Review by laurkim — APPROVED** [review]

**Pull request merged**